pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -336,6 +340,18 @@ func (r *ReconcileHyperConverged) doReconcile(req *common.HcoRequest) (reconcile
 	knownHcoVersion, _ := req.Instance.Status.GetVersion(hcoVersionName)
 
 	if !r.upgradeMode && !init && knownHcoVersion != r.ownVersion {
+		if hcoutil.GetClusterInfo().IsManagedByOLM() {
+			// Ensure OLM is not interfering while we are upgrading components.
+			if err := setOperatorCondition(req.Ctx, r.client, metav1.Condition{
```

Comment:
is the setOperatorCondition will be a method in the Reconciler type, there no need to pass the client and the version as parameters.